src/rawMD5_fmt_plug.c

```diff
@@ -187,9 +187,8 @@ static char *split(char *ciphertext, int index, struct fmt_main *self)
 {
 	static char out[TAG_LENGTH + CIPHERTEXT_LENGTH + 1] = FORMAT_TAG;
 
-	if (ciphertext[0] == '$' &&
-	    !strncmp(ciphertext, FORMAT_TAG, TAG_LENGTH))
```

Comment:
IIRC, the performance improvement, while minimal, was non-negligible on multi-million hash files. So I'd prefer that we keep the separate first character check, and maybe introduce it to other `raw-md5*` formats as well. OK to add it to `raw-md4` too.

It is really unfortunate that the lower-casing itself will have a much bigger performance impact, though. But it looks like we'll have to accept it for now.

Reply:
So... should I add it to all three files I touched within this PR?

Reply:
Please feel free to do that, or at least don't remove it from `raw-md5`, which is presumably the most frequently used one of the three.